File: project/backend/src/main/java/caaruujuuwoo65/backend/model/Review.java
package caaruujuuwoo65.backend.model;

import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.time.LocalDateTime;

@Entity
@Table(name = "reviews")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class Review {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long reviewId;

    private Integer rating;

    private String review;

    private LocalDateTime createdAt;

    private LocalDateTime updatedAt;

    @ManyToOne
    @JoinColumn(name = "product")
    @JsonIgnore
    private Product product;

    @ManyToOne
    @JoinColumn(name = "user")
    private User user;
}

File: project/frontend/src/types/Review.ts
import { UserReviewInfo } from "./UserReviewInfo";
import { Product } from "./product/Product";

export type Review = {
    reviewId: number;
    rating: number;
    review: string;
    createdAt: Date;
    updatedAt: Date;
    product: Product;
    user: UserReviewInfo;
}
